src/tool/hpcrun/sample-sources/nvidia.c

```diff
@@ -412,8 +404,8 @@ METHOD_FN(process_event_list, int lush_metrics)
 
   // Register cupti callbacks
   cupti_init();
-  cupti_callbacks_subscribe();
   cupti_start();
```

Comment:
Why do we subscribe after starting cupti? Is this an equivalent choice? If so, I prefer subscribe before start. If not equivalent, it needs a comment.

Reply:
Other than my comment, this looks good. One other question: is the prior race eliminated or just less likely?

Reply:
Oh, it does not matter where to put cupti_callbacks_subscribe and cupti_start. They are equivalent.

Reply:
The prior race is eliminated completely as far as I can tell.